frontend/view-edit-client/interactions/single-interaction-slat.component.tsx

```diff
@@ -335,14 +331,95 @@ export default React.forwardRef<any, SingleClientInteractionProps>(
         </div>
       </div>
     );
+
+    function updateInteractionKind(evt: React.ChangeEvent<HTMLInputElement>) {
+      setInteractionKind(evt.target.value as InteractionKind);
+    }
+
+    function cuServiceSelect() {
+      return (
+        <select
+          ref={ref}
+          value={
+            selectedService
+              ? selectedService.type + selectedService.service.id
+              : ""
+          }
+          onChange={(evt) => {
+            const serviceId = Number(evt.target.value.slice("CU".length));
+            setSelectedService({
+              type: "CU",
+              service: services.find((s) => s.id === serviceId),
+            });
+          }}
+          aria-labelledby={`provided-service-${props.interactionIndex}`}
+          className="services-select"
+          name={`provided-service-${props.interactionIndex}`}
+          required
+        >
+          <option value="" disabled hidden>
+            Choose here
+          </option>
+          {Object.keys(groupedServices).map((programName) => (
+            <optgroup label={programName} key={programName}>
+              {groupedServices[programName].map((service) => (
+                <option key={service.id} value={"CU" + service.id}>
+                  {service.serviceName}
+                </option>
+              ))}
+            </optgroup>
+          ))}
+        </select>
+      );
+    }
+
+    function referralSelect() {
+      return (
+        <select
+          ref={ref}
+          value={
+            selectedService
+              ? selectedService.type + selectedService.service.id
+              : ""
+          }
+          onChange={(evt) => {
+            const serviceId = Number(evt.target.value.slice("Partner".length));
+            setSelectedService({
+              type: "Partner",
+              service: partnerServices.find((ps) => ps.id === serviceId),
+            });
+          }}
+          aria-labelledby={`provided-service-${props.interactionIndex}`}
+          className="services-select"
+          name={`provided-service-${props.interactionIndex}`}
+          required
+        >
+          <option value="" disabled hidden>
+            Choose here
+          </option>
+          {props.partnersResponse.map((partner) => (
```

Comment:
Seems to be an empty array when passed through to this component

Reply:
On your local environment, the partners will be empty until you create one. This is done in the Admin Settings page. But once you create a partner, this component will start out with an empty array but then update to have the data from the db

Reply:
Ah gotcha, my bad I thought it auto populates similar to services. Forgot referrals was a new feature!!